### PR TITLE
Adding winget workflow

### PR DESCRIPTION
**Pull request by ldennington**
Adding a shiny new workflow to publish releases to winget! 🥳

I validated this workflow locally by successfully opening a [PR](https://github.com/ldennington/winget-playground/pull/4) against my winget-playground repo and running `winget validate` to ensure the generated manifest adheres to winget's required schema. Any feedback on the fields I've included (\*cough\* `ShortDescription:` \*cough\*) or suggestions for additional fields/values are welcome!

### CODE REVIEW

File: .github/workflows/release-winget.yaml
Comment (derrickstolee) [inline .github/workflows/release-winget.yaml]: Did I mis-remember that this was `v1.2` in your GCM Core one?

File: .github/workflows/release-winget.yaml
Comment (derrickstolee) [inline .github/workflows/release-winget.yaml]: I want to call attention to this bit, as it will be the forward-facing description to users browsing the `winget` listing.

I think my only comment is to replace `Note -` with `Note:`.

File: .github/workflows/release-winget.yaml
Comment (derrickstolee) [reply .github/workflows/release-winget.yaml]: [Yes, you used `v1.2`](https://github.com/microsoft/Git-Credential-Manager-Core/pull/332/files#diff-b4ee64863401eef676b49220519b9c6b6532f995c8b256ed6273b71237438fcaR12)

Comment (ldennington) [reply .github/workflows/release-winget.yaml]: Oh yikes, I can't believe I forgot to do that. Totally my bad.

File: .github/workflows/release-winget.yaml
Comment (ldennington) [reply .github/workflows/release-winget.yaml]: I originally had it as `Note:` but ran into some issues with `winget validate`. I'm sure there's an easy way to escape though, I'll experiment a bit.

Comment (ldennington) [reply .github/workflows/release-winget.yaml]: Nvm, I lied, it seems fine when we have a multi-line description like this.

**Review by derrickstolee — APPROVED** [review]
I think this is ready to merge. We could consider adding a longer `Description:` text, but that can wait (if we even think it is necessary).

**Comment by ldennington** [comment]
> I think this is ready to merge. We could consider adding a longer `Description:` text, but that can wait (if we even think it is necessary).

Yay!!! So exciting! I'll wait until checks have completed then merge.

**Comment by derrickstolee** [comment]
Apparently the builds have been updated to not re-run on a tree where the build is known to succeed, but instead of reporting "success" it reports "skipped". I will merge this PR without the green builds.

@dscho: are you aware of this issue? I know you have a lot of experience with these builds.

**Comment by ldennington** [comment]
> Apparently the builds have been updated to not re-run on a tree where the build is known to succeed, but instead of reporting "success" it reports "skipped". I will merge this PR without the green builds.
> 
> @dscho: are you aware of this issue? I know you have a lot of experience with these builds.

Thank you!!

**Comment by dscho** [comment]
> Apparently the builds have been updated to not re-run on a tree where the build is known to succeed, but instead of reporting "success" it reports "skipped". I will merge this PR without the green builds.
> 
> @dscho: are you aware of this issue? I know you have a lot of experience with these builds.

I am the reason for this: 7d78d5fc1a91b683dde970e5e48b6d9a873cfd99. It is too bad that we cannot make the branch protection dependent on the overall workflow. In my mind, it is a shortcoming of GitHub Actions that you can only require individual jobs to pass...

**Pull request merged**
